django/db/models/expressions.py

```diff
@@ -858,7 +858,10 @@ class ExpressionWrapper(Expression):
 
     def __init__(self, expression, output_field):
         super().__init__(output_field=output_field)
-        if getattr(expression, '_output_field_or_none', True) is None:
+        if (
+            isinstance(expression, Expression) and
+            getattr(expression, '_output_field_or_none', None) is None
+        ):
```

Comment:
I haven't found a better way to distinct between `expression`\`s that don't have `_output_field_or_none` (e.g. `Q`) and `expression`\`s that have `_output_field_or_none` but it raises an `AttributeError` (e.g. `CombinedExpression` with `F`).